src/piral-build.test.ts

```diff
@@ -7,63 +7,192 @@ runTests('piral-build', ({ test, setup }) => {
+        return JSON.stringify(packageJson, undefined, 2);
+      },
+    });
+
+    await ctx.run(`npx piral build --type emulator`);
+
+    await ctx.assertFiles({
+      'dist/emulator/*.tgz'(files: Array<string>) {
+        expect(files).toHaveLength(1);
+        ctx.setRef('emulator', files[0]);
+      },
+    });
+  });
+
+  test('emulator-sources-standard-template', 'can produce emulator sources build', ['build.piral'], async (ctx) => {
+    await ctx.run(`npx piral build --type emulator-sources`);
+
+    await ctx.assertFiles({
+      'package-lock.json': true,
+      'package.json'(content: string) {
+        expect(content).toContain('"pilets": {');
+      },
+      'node_modules/piral-cli/package.json': true,
+      'emulator/package.json': true,
+      'emulator/app/index.html'(content: string) {
+        expect(content).not.toBe('');
+        expect(content).toContain('<title>My Piral Instance</title>');
+        expect(content).toContain('<script defer src="/index.b68111.js"></script>');
+      },
+    });
+  });
+
+  test(
+    'some-invalid-type-standard-template',
+    'can not produce build with invalid type',
+    ['build.piral'],
+    async (ctx) => {
+      try {
+        await ctx.run(`npx piral build --type some-invalid-type-standard-template`);
+      } catch {}
+
+      await ctx.assertFiles({
+        'package.json': true,
+        'tsconfig.json': true,
+        'node_modules/piral-cli/package.json': true,
+        'src/index.html': true,
+        'package-lock.json': true,
+      });
+    },
+  );
+
   test(
-    'release-standard-template',
-    'can produce a release build',
+    'release-standard-template-with-default-public-url',
+    'can produce a release build with default public url',
     ['build.piral'],
     async (ctx) => {
-      await ctx.run(`npx piral build --type release`);
+      await ctx.run(`npx piral build --type release --public-url '/'`);
 
       await ctx.assertFiles({
         'dist/release/index.html'(content: string) {
           expect(content).not.toBe('');
+          expect(content).toContain('<script defer="defer" src="/index.');
         },
       });
     },
   );
 
   test(
-    'emulator-standard-template',
-    'can produce an emulator build',
+    'release-standard-template-with-a-different-public-url',
+    'can produce a release build with a different public url ./',
     ['build.piral'],
     async (ctx) => {
-      await ctx.setFiles({
-        'package.json'(content: string) {
-          const packageJson = JSON.parse(content);
-          packageJson.name = 'app-shell';
-          return JSON.stringify(packageJson, undefined, 2);
+      await ctx.run(`npx piral build --type release --public-url 'different-public-url/'`);
+
+      await ctx.assertFiles({
+        'dist/release/index.html'(content: string) {
+          expect(content).not.toBe('');
+          expect(content).toContain('<script defer="defer" src="/different-public-url/index.');
```

Comment:
We don't know about that `defer` here - it's too specific. I suggest you to check via the `src` only (we also don't know ab out the `index`, e.g., for esbuild the file is called main.js)